hashintel/hash
# Add timeout for Collab position polling

## 🌟 What is the purpose of this PR?




This PR adds a timeout to position polling for Collab, such that timeouts in other parts of the stack (cloudflare) don't end up causing errors. Previously cloudflare would respond with a 524 response code that included no CORS headers. This doesn't fix any of the broader errors in collab, but is definitely a source of errors.

## 🔗 Related links





- [related Asana task](https://app.asana.com/0/1200211978612931/1202518348357256/f) _(internal)_

## 🚫 Blocked by





## 🔍 What does this change?




- Adds a timeout in collab position polling
- Skips setting positions if timeout is reached in the frontend

## 📜 Does this require a change to the docs?



- Nope

## ⚠️ Known issues


- This may not be the original bug from the task, will try to recreate that. 

## 🐾 Next steps


- Fix other instability issues

## 🛡 What tests cover this?




## ❓ How to test this?



1.  Checkout the branch and start the frontend/backend
1.  Open any page
1.  Confirm that the `/positions?poll=true` requests have a max duration of 5 seconds

## 📹 Demo


![image](https://user-images.githubusercontent.com/6988668/180439135-1db956b8-e7b1-4d53-9f8d-a175fc76b319.png)
